src/pages/workspace/accounting/netsuite/advanced/NetSuiteAdvancedPage.tsx

```diff
@@ -0,0 +1,286 @@
+import {View} from 'react-native';
+import ConnectionLayout from '@components/ConnectionLayout';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as Connections from '@libs/actions/connections/NetSuiteCommands';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import type {DividerLineItem, MenuItem, ToggleItem} from '@pages/workspace/accounting/netsuite/types';
+import type {WithPolicyConnectionsProps} from '@pages/workspace/withPolicyConnections';
+import withPolicyConnections from '@pages/workspace/withPolicyConnections';
+import ToggleSettingOptionRow from '@pages/workspace/workflows/ToggleSettingsOptionRow';
+import * as Policy from '@userActions/Policy/Policy';
+import CONST from '@src/CONST';
+
+function NetSuiteAdvancedPage({policy}: WithPolicyConnectionsProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const policyID = policy?.id ?? '-1';
+
+    const config = policy?.connections?.netsuite?.options.config;
+    const autoSyncConfig = policy?.connections?.netsuite?.config;
+
+    const {payableList} = policy?.connections?.netsuite?.options?.data ?? {};
+
+    const selectedReimbursementAccount = useMemo(
+        () => (payableList ?? []).find((payableAccount) => payableAccount.id === config?.reimbursementAccountID),
+        [payableList, config?.reimbursementAccountID],
+    );
+    const selectedCollectionAccount = useMemo(() => (payableList ?? []).find((payableAccount) => payableAccount.id === config?.collectionAccount), [payableList, config?.collectionAccount]);
+    const selectedApprovalAccount = useMemo(() => {
+        if (config?.approvalAccount === CONST.NETSUITE_APPROVAL_ACCOUNT_DEFAULT) {
+            return {
+                id: CONST.NETSUITE_APPROVAL_ACCOUNT_DEFAULT,
+                name: translate('workspace.netsuite.advancedConfig.defaultApprovalAccount'),
+            };
+        }
+        return (payableList ?? []).find((payableAccount) => payableAccount.id === config?.approvalAccount);
+    }, [config?.approvalAccount, payableList, translate]);
+
+    const menuItems: Array<MenuItem | ToggleItem | DividerLineItem> = [
+        {
+            type: 'toggle',
+            title: translate('workspace.accounting.autoSync'),
+            subtitle: translate('workspace.netsuite.advancedConfig.autoSyncDescription'),
+            isActive: !!autoSyncConfig?.autoSync.enabled,
+            switchAccessibilityLabel: translate('workspace.netsuite.advancedConfig.autoSyncDescription'),
+            shouldPlaceSubtitleBelowSwitch: true,
+            onCloseError: () => Policy.clearNetSuiteAutoSyncErrorField(policyID),
+            onToggle: () => Connections.updateNetSuiteAutoSync(policyID, !autoSyncConfig?.autoSync.enabled, !!autoSyncConfig?.autoSync.enabled),
```

Comment:
Should we rely on Onyx or the UI for the toggle value. I think we should use `onToggle(isEnabled: boolean)` to pass as a arguments.

Reply:
+1 to this